src/crypto-providers/util.ts

```diff
@@ -453,22 +474,52 @@ const isDeviceVersionGTE = (
     && deviceVersion.patch >= thresholdVersion.patch
   )
 
+const formatVotingRegistrationMetaData = (
+  votingPublicKey: Buffer,
+  stakePub: Buffer,
+  address: Buffer,
+  nonce: BigInt,
+  signature: Buffer,
+): VotingRegistrationMetaData => [
+  new Map<number, Map<number, Buffer | BigInt>>([
+    [
+      61284,
+      new Map<number, Buffer | BigInt>([
+        [1, votingPublicKey],
+        [2, stakePub],
+        [3, address],
+        [4, nonce],
+      ]),
+    ],
+    [
+      61285,
+      new Map<number, Buffer>([
+        [1, signature],
+      ]),
+    ],
+  ]),
+  [],
```

Comment:
this actually isn't the metadata but the whole auxiliary data if you check the CDDL: https://github.com/input-output-hk/cardano-ledger-specs/blob/49d29c3bd07bd554cc8d967a2eb7b318324f0cff/shelley-ma/shelley-ma-test/cddl-files/shelley-ma.cddl#L214

which is fine for verifying it's hash, but bear in mind cardano-cli takes the metadata CBOR (i.e. only the CBOR serialization of the `Map<number, Map<number, Buffer | BigInt>>`) and appends the empty list to list in the final transaction. You can find further context here: https://github.com/input-output-hk/cardano-node/issues/2479

Reply:
also the returned file should be in binary to be a valid file for the --metadata-cbor-file flag of cardano-cli's command to build the raw transaction

Reply:
it also writes it has the hex in the file instead of binary cbor.

Reply:
Fixed, tried to create transaction in `cardano-cli`, seems to be working.